scripts/core/exp-command-buffer.yml

```diff
@@ -588,6 +588,75 @@ returns:
     - $X_RESULT_ERROR_OUT_OF_RESOURCES
 --- #--------------------------------------------------------------------------
 type: function
+desc: "Append a USM Prefetch command to a command-buffer object"
+class: $xCommandBuffer
+name: AppendUSMPrefetchExp
+params:
+    - type: $x_exp_command_buffer_handle_t
+      name: hCommandBuffer
+      desc: "[in] handle of the command-buffer object."
+    - type: "void*"
+      name: pMemory
+      desc: "[in] pointer to USM allocated memory to prefetch."
+    - type: "size_t"
+      name: size
+      desc: "[in] size in bytes to be fetched."
+    - type: $x_usm_migration_flags_t
+      name: flags
+      desc: "[in] USM prefetch flags"
+    - type: uint32_t
+      name: numSyncPointsInWaitList
+      desc: "[in] The number of sync points in the provided dependency list."
+    - type: "const $x_exp_command_buffer_sync_point_t*"
+      name: pSyncPointWaitList
+      desc: "[in][optional] A list of sync points that this command depends on."
+    - type: "$x_exp_command_buffer_sync_point_t*"
+      name: pSyncPoint
+      desc: "[out][optional] sync point associated with this command."
+returns:
+    - $X_RESULT_ERROR_INVALID_COMMAND_BUFFER_EXP
+    - $X_RESULT_ERROR_INVALID_COMMAND_BUFFER_SYNC_POINT_EXP
+    - $X_RESULT_ERROR_INVALID_COMMAND_BUFFER_SYNC_POINT_WAIT_LIST_EXP:
+        - "`pSyncPointWaitList == NULL && numSyncPointsInWaitList > 0`"
+        - "`pSyncPointWaitList != NULL && numSyncPointsInWaitList == 0`"
+    - $X_RESULT_ERROR_INVALID_MEM_OBJECT
+    - $X_RESULT_ERROR_INVALID_SIZE:
+        - "`size == 0`"
+        - "If `size` is higher than the allocation size of `pMemory`"
+    - $X_RESULT_ERROR_OUT_OF_HOST_MEMORY
+    - $X_RESULT_ERROR_OUT_OF_RESOURCES
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Append a USM Advise command to a command-buffer object"
+class: $xCommandBuffer
+name: AppendUSMAdviseExp
+params:
+    - type: $x_exp_command_buffer_handle_t
+      name: hCommandBuffer
+      desc: "[in] handle of the command-buffer object."
+    - type: "void*"
+      name: pMemory
+      desc: "[in] pointer to the USM memory object."
+    - type: "size_t"
+      name: size
+      desc: "[in] size in bytes to be advised."
+    - type: $x_usm_advice_flags_t
+      name: advice
+      desc: "[in] USM memory advice"
+    - type: "$x_exp_command_buffer_sync_point_t*"
+      name: pSyncPoint
+      desc: "[out][optional] sync point associated with this command."
```

Comment:
It's interesting to me that there is no dependency list parameters here, i.e.
```
    - type: uint32_t
      name: numSyncPointsInWaitList
      desc: "[in] The number of sync points in the provided dependency list."
    - type: "const $x_exp_command_buffer_sync_point_t*"
      name: pSyncPointWaitList
      desc: "[in][optional] A list of sync points that this command depends on."
    - type: "$x_exp_command_buffer_sync_point_t*"
 ```

I'm aware that regular [urEnqueueUSMAdvise](https://oneapi-src.github.io/unified-runtime/core/api.html#ur__api_8h_1a837f634bdc885d96fc7d56548b47c1dd) doesn't have these, probably because [zeCommandListAppendMemAdvise](https://spec.oneapi.io/level-zero/latest/core/api.html?#zecommandlistappendmemadvise) doesn't. However, `clEnqueueMemAdviseINTEL` in the [OpenCL USM extension](https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_unified_shared_memory.html) does have a dependency list, and a SYCL command-group with a `mem_advise` can express dependencies.

Reply:
As matching https://github.com/oneapi-src/unified-runtime/pull/887, we should add the dependency list

Reply:
The dependency list has been added to this function.